index/new.go

```diff
@@ -0,0 +1,50 @@
+package index
+
+import (
+	"path/filepath"
+
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+
+	"github.com/buildpacks/imgutil"
+)
+
+// NewIndex will return a New Empty ImageIndex that can be modified and saved to a registry
+func NewIndex(repoName string, ops ...Option) (idx *ImageIndex, err error) {
+	var idxOps = &imgutil.IndexOptions{}
+	for _, op := range ops {
+		if err = op(idxOps); err != nil {
+			return idx, err
+		}
+	}
+
+	if err = ValidateRepoName(repoName, idxOps); err != nil {
+		return idx, err
+	}
+
+	layoutPath := filepath.Join(idxOps.XdgPath, imgutil.MakeFileSafeName(repoName))
+
+	var cnbIndex *imgutil.CNBIndex
+	switch idxOps.Format {
+	case types.DockerManifestList:
+		cnbIndex, err = imgutil.NewCNBIndex(imgutil.NewEmptyDockerIndex(), *idxOps)
+		if err != nil {
+			return idx, err
+		}
+		// TODO I don't think we should write into disk during creation
+		_, err = layout.Write(layoutPath, imgutil.NewEmptyDockerIndex())
```

Comment:
I already started removing this logic, I did it for the layout package, I will do the same for local and remote

Reply:
Actually, the [empty.Index](https://github.com/google/go-containerregistry/blob/051d642e295e3e86b58bb5feabb184036fa34a52/pkg/v1/empty/index.go#L51-L57) has a couple of methods that always returns an error that we often tend to call, So I thought writing it to dist and loading with [layout](https://github.com/google/go-containerregistry/blob/051d642e295e3e86b58bb5feabb184036fa34a52/pkg/v1/layout/index.go#L84-L123) might work like a hack to fix these